### PR TITLE
feat(plugin-infra): add metadata previewer API; add minimal mode

### LINKED ISSUE #3687: [Demand] Plugin infra: metadata render

### PR DESCRIPTION
**Pull request by Jack-Works**
closes #3687

This PR adds a new plugin infra API to simplify and empower the ability of rendering metadata in plugins.

This is still in design. The current API looks like this:

```ts
import type { Plugin } from '@masknet/plugin-infra'
import { forwardRef, useImperativeHandle } from 'react'
import { Ok, Err } from 'ts-results'

const metadataReader: Plugin.ContextFree.MetadataRender.MetadataReader<unknown> = (meta) => {
    const raw = meta?.get('io.mask.example/v1')
    if (raw) return Ok(raw)
    return Err.EMPTY
}
const render: Plugin.ContextFree.MetadataRender.StaticRenderComponent<unknown> = forwardRef((props, ref) => {
    useImperativeHandle(ref, () => ({}), [])
    return <>Metadata render for key "io.mask.example/v1" {JSON.stringify(props.metadata)}</>
})

const contextFree: Plugin.ContextFree.DefinitionDeferred = {
    metadataRender: new Map([[metadataReader, render]]),
}
export default contextFree
```

### CODE REVIEW

**Review by guanbinrui — DISMISSED** [review]

**Comment by yanzhihong23** [comment]
@wenluomask

**Comment by wenluomask** [comment]
Refer to Jira: https://mask.atlassian.net/browse/MASK-391

**Comment by wenluomask** [comment]
Checked.

File: packages/mask/src/plugins/Pets/SNSAdaptor/animate.tsx [hunk 1]
```diff
@@ -36,13 +39,14 @@ const AnimatePic = () => {
 
     const [show, setShow] = useState(false)
     const [infoShow, setInfoShow] = useState(false)
+    const enabled = useValueRef(currentPluginMinimalModeNOTEnabled['plugin:' + PetsPluginID])
```
Comment (Jack-Works) [inline packages/mask/src/plugins/Pets/SNSAdaptor/animate.tsx:42]: `useIsMinimalMode`. Do not access the raw settings directly.

File: packages/mask/src/components/DataSource/usePersonaConnectStatus.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+export function usePersonaConnectStatus() {
+    const personas = useMyPersonas()
+    const lastRecognized = useLastRecognizedIdentity()
+
+    const createPersona = () => {
+        Services.Welcome.openOptionsPage(DashboardRoutes.Setup)
+    }
+
+    const connectPersona = async () => {
```
Comment (Jack-Works) [inline packages/mask/src/components/DataSource/usePersonaConnectStatus.ts:19]: Please move this function top level.

File: packages/mask/src/components/InjectedComponents/DecryptedPost/DecryptPostFailed.tsx [hunk 3]
```diff
@@ -16,10 +17,12 @@ export interface DecryptPostFailedProps {
...
 }
 export const DecryptPostFailed = memo(function DecryptPostFailed(props: DecryptPostFailedProps) {
-    const { AdditionalContentProps, NotSetupYetPromptProps, author, postedBy, error } = props
+    const { AdditionalContentProps, author, postedBy, error } = props
     const { t } = useI18N()
-    if (error?.message === DecryptFailedReason.MyCryptoKeyNotFound)
-        return <NotSetupYetPrompt {...NotSetupYetPromptProps} description="decryptPostFailed" />
+
+    if (error?.message === DecryptFailedReason.MyCryptoKeyNotFound) {
+        return <MaskPluginWrapper pluginName="" />
```
Comment (Jack-Works) [inline packages/mask/src/components/InjectedComponents/DecryptedPost/DecryptPostFailed.tsx:24]: Why you'd reveal a MaskPluginWrapper when the decryption failed?

File: packages/mask/src/plugins/MaskPluginWrapper.tsx [hunk 4]
```diff
@@ -26,38 +32,96 @@ const useStyles = makeStyles()((theme) => {
...
+            },
         },
     }
 })
 
 export default function MaskPluginWrapper(props: PluginWrapperProps) {
     const { classes } = useStyles()
-    const { pluginName, children } = props
+    const { pluginName, children, action, publisher } = props
+    const personaConnectStatus = usePersonaConnectStatus()
```
Comment (Jack-Works) [inline packages/mask/src/plugins/MaskPluginWrapper.tsx:68]: You should not mix persona connect logic into this component.

If you only want to reuse the UI, please split the UI as a separate component.

**Comment by Jack-Works** [comment]
@yanzhihong23 here is some review

File: packages/mask/src/plugins/MaskPluginWrapper.tsx (hunk 4 above)
Comment (Jack-Works) [reply packages/mask/src/plugins/MaskPluginWrapper.tsx:68]: @yanzhihong23 this one

File: packages/mask/src/plugins/MaskPluginWrapper.tsx [hunk 5]
```diff
@@ -26,38 +32,96 @@ const useStyles = makeStyles()((theme) => {
...
+    const { t } = useI18N()
+
+    const renderChildren = useMemo(() => {
+        return personaConnectStatus.connected && children
+    }, [personaConnectStatus, children])
+
+    const name = useMemo(() => {
```
Comment (Jack-Works) [inline packages/mask/src/plugins/MaskPluginWrapper.tsx]: No need to memo `name` and `renderChildren`.

File: packages/mask/src/plugins/MaskPluginWrapper.tsx [hunk 6]
```diff
@@ -26,38 +32,96 @@ const useStyles = makeStyles()((theme) => {
...
+        return !personaConnectStatus.hasPersona
+            ? t('please_create_persona')
+            : !personaConnectStatus.connected
+            ? t('please_connect_persona')
+            : pluginName
+    }, [personaConnectStatus, pluginName])
+
+    const actionButton = useMemo(() => {
```
Comment (Jack-Works) [inline packages/mask/src/plugins/MaskPluginWrapper.tsx:77]: If you don't add `t` to dependency, it will not re-render when the language changes.

**Pull request merged**